### DISCUSSION TITLE
Adding new meeting types

### DISCUSSION THREAD

**Discussion by eflashlogistics** [code4recovery/12-step-meeting-list#869] [Q&A]
Greetings. I added to the functions.php for two new meeting types. I am only getting one. Is there a proper way to write this statement? 

if (function_exists('tsml_custom_types')) {
    tsml_custom_types(array(
        'XYZ' => 'Young Adults',
    ));
}
if (function_exists('tsml_custom_types')) {
    tsml_custom_types(array(
        'XYZ' => 'Literature Study',
    ));
}

**Comment by Grateful-Viv** [comment]
Looks like the second XYZ is overwriting the first one. 

Maybe use YA for young adults and LS for literature study?

**Reply by eflashlogistics** [reply]
Yes! I didnt really understand what I was doing there. 

Question, how do I pull up a list of those codes? And can I remove the unneeded ones?

**Comment by brianw-area24** [comment]
If you're doing an AA web site, then I recommend using the existing types whenever possible, because they match what is used by the Meeting Guide Application, or you could get unintended consequences. Best case, your types just won't show up Meeting Guide, and worst case, they will have and incorrect type in Meeting Guide. For example using LS for 'Literature Study' would go into Meeting Guide as 'Living Sober', because LS is a predefined type. On way to prevent this is to us something like MY as a prefix, so you'd have MYYA and MYLS. If you're not going connect with Meeting Guide, or have your meetings imported by other sites, then this isn't as much of a consideration

As for hiding types, the form for adding/editing meetings should only show the meeting types that are being used on your site, so effectively, the rest are hidden. There is a link on the form where you can toggle between seeing or not seeing the types that aren't in use. Also, the dropdown field on the meetings page should only show types that being used.

The codes used by the different fellowships are in code, probably the easiest place to see them are on GitHub, at https://github.com/code4recovery/12-step-meeting-list/blob/main/includes/variables.php, starting at line 571, continuing through the end of the file.

**Reply by eflashlogistics** [reply]
Thank you! Its for al-anon so no meeting guide. 

I think the curious part of me wants to know all the codes. Theres a couple I want to overite though like smoking permitted. I am in california, in San Luis Obispo county, where they banned smoking in restaurants first. 

I am scared of these old ladies and if they see it as an option 😂

**Reply by brianw-area24** [reply]
Here's the code for al-anon:
https://github.com/code4recovery/12-step-meeting-list/blob/d1034c316fd2ec05b7c7e17f81916ce5befb45a5/includes/variables.php#L601-L634


I think this code would remove Smoking Permitted:
if (function_exists('tsml_custom_types')) {
	unset($tsml_programs['al-anon']['types']['SM']);
}
